### PR TITLE
Booleand validator

### PR DESCRIPTION
**Pull request by andrechi1**
Implement the boolean True and False validator.

### CODE REVIEW

**Comment by JeremySkinner** [comment]
Hi there, thanks for the suggestion. 

This functionality is already covered bu the existing `Equal` validator, eg `RuleFor(x => x.IsValid).Equal(true)`. I don't see a use case for having explicit `TrueValidator` and `FalseValidator` classes for this, when the existing `EqualValidator` works fine for this scenario.

If you really want to have explicit `True` and `False` methods, you can implement these as extension methods that wrap the existing `Equal` method. No need for new validator classes. Eg:

```
public static IRuleBuilderOptions<T, bool> True<T>(this IRuleBuilder<T, bool> ruleBuilder) {
	return ruleBuilder.Equal(true);
}

public static IRuleBuilderOptions<T, bool?> True<T>(this IRuleBuilder<T, bool?> ruleBuilder) {
	return ruleBuilder.Equal(true);
}
		
public static IRuleBuilderOptions<T, bool> False<T>(this IRuleBuilder<T, bool> ruleBuilder) {
	return ruleBuilder.Equal(false);
}

public static IRuleBuilderOptions<T, bool?> False<T>(this IRuleBuilder<T, bool?> ruleBuilder) {
	return ruleBuilder.Equal(false);
}
		
```

**Comment by andrechi1** [comment]
It's a small contribution, but it becomes clearer.

What to create TrueValidator and FalseValidato, copy it as implemented NullValidator.

File: src/FluentValidation.Tests/BooleanValidatorTests.cs [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+// distributed under the License is distributed on an "AS IS" BASIS, 
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. 
+// See the License for the specific language governing permissions and 
+// limitations under the License.
+// 
+// The latest version of this file can be found at https://github.com/jeremyskinner/FluentValidation
+#endregion
+
+namespace FluentValidation.Tests
+{
```
Comment (JeremySkinner) [inline src/FluentValidation.Tests/BooleanValidatorTests.cs]: Please ensure that contributions match the rest of the project coding style - all braces should be at the end of the line.

File: src/FluentValidation.Tests/BooleanValidatorTests.cs [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+	using Xunit;
+
+
```
Comment (JeremySkinner) [inline src/FluentValidation.Tests/BooleanValidatorTests.cs]: Remove unnecessary whitespace

File: src/FluentValidation.Tests/BooleanValidatorTests.cs [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+		[Fact]
+		public void When_the_value_is_true_the_validator_should_pass() {
+			var validator = new TestValidator(v => v.RuleFor(x => x.IsValid).True());
+			var result = validator.Validate(new Person { IsValid = true});
+			result.IsValid.ShouldBeTrue();
+		}
+
+		[Fact]
+		public void When_the_value_is_not_true_the_validator_not_should_pass()
+		{
```
Comment (JeremySkinner) [inline src/FluentValidation.Tests/BooleanValidatorTests.cs]: Braces at end of line please.

File: src/FluentValidation.Tests/BooleanValidatorTests.cs [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+			var validator = new TestValidator(v => v.RuleFor(x => x.IsValid).True());
+			var result = validator.Validate(new Person { IsValid = false });
+			result.IsValid.ShouldBeFalse();
+		}
+
+		[Fact]
+		public void When_the_value_is_false_the_validator_should_pass()
+		{
```
Comment (JeremySkinner) [inline src/FluentValidation.Tests/BooleanValidatorTests.cs]: Braces at end of line please.

File: src/FluentValidation.Tests/BooleanValidatorTests.cs [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+			var validator = new TestValidator(v => v.RuleFor(x => x.IsValid).False());
+			var result = validator.Validate(new Person { IsValid = false });
+			result.IsValid.ShouldBeTrue();
+		}
+
+		[Fact]
+		public void When_the_value_is_true_the_validator_not_should_pass()
+		{
```
Comment (JeremySkinner) [inline src/FluentValidation.Tests/BooleanValidatorTests.cs]: Braces at end of line please.

File: src/FluentValidation.Tests/BooleanValidatorTests.cs [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+			var validator = new TestValidator(v => v.RuleFor(x => x.IsValid).False());
+			var result = validator.Validate(new Person { IsValid = true });
+			result.IsValid.ShouldBeFalse();
+		}
+
+		[Fact]
+		public void When_the_nullable_value_is_true_the_validator_should_pass()
+		{
```
Comment (JeremySkinner) [inline src/FluentValidation.Tests/BooleanValidatorTests.cs]: Braces at end of line please.

File: src/FluentValidation.Tests/BooleanValidatorTests.cs [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+			var validator = new TestValidator(v => v.RuleFor(x => x.NullableIsValid).True());
+			var result = validator.Validate(new Person { NullableIsValid = true });
+			result.IsValid.ShouldBeTrue();
+		}
+
+		[Fact]
+		public void When_the_nullable_value_is_not_true_the_validator_not_should_pass()
+		{
```
Comment (JeremySkinner) [inline src/FluentValidation.Tests/BooleanValidatorTests.cs]: Braces at end of line please.

File: src/FluentValidation.Tests/BooleanValidatorTests.cs [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+			var validator = new TestValidator(v => v.RuleFor(x => x.NullableIsValid).True());
+			var result = validator.Validate(new Person { NullableIsValid = false });
+			result.IsValid.ShouldBeFalse();
+		}
+
+		[Fact]
+		public void When_the_nullable_value_is_false_the_validator_should_pass()
+		{
```
Comment (JeremySkinner) [inline src/FluentValidation.Tests/BooleanValidatorTests.cs]: Braces at end of line please.

File: src/FluentValidation.Tests/BooleanValidatorTests.cs [hunk 9]
```diff
@@ -0,0 +1,108 @@
...
+			var validator = new TestValidator(v => v.RuleFor(x => x.NullableIsValid).False());
+			var result = validator.Validate(new Person { NullableIsValid = false });
+			result.IsValid.ShouldBeTrue();
+		}
+
+		[Fact]
+		public void When_the_nullable_value_is_true_the_validator_not_should_pass()
+		{
```
Comment (JeremySkinner) [inline src/FluentValidation.Tests/BooleanValidatorTests.cs]: Braces at end of line please.

File: src/FluentValidation.Tests/BooleanValidatorTests.cs [hunk 10]
```diff
@@ -0,0 +1,108 @@
...
+			var validator = new TestValidator(v => v.RuleFor(x => x.NullableIsValid).False());
+			var result = validator.Validate(new Person { NullableIsValid = true });
+			result.IsValid.ShouldBeFalse();
+		}
+
+		[Fact]
+		public void When_the__true_boolean_nullable_value_validator_not_should_pass()
+		{
```
Comment (JeremySkinner) [inline src/FluentValidation.Tests/BooleanValidatorTests.cs]: Braces at end of line please.

File: src/FluentValidation.Tests/BooleanValidatorTests.cs [hunk 11]
```diff
@@ -0,0 +1,108 @@
...
+		[Fact]
+		public void When_the_nullable_value_is_true_the_validator_not_should_pass()
+		{
+			var validator = new TestValidator(v => v.RuleFor(x => x.NullableIsValid).False());
+			var result = validator.Validate(new Person { NullableIsValid = true });
+			result.IsValid.ShouldBeFalse();
+		}
+
+		[Fact]
+		public void When_the__true_boolean_nullable_value_validator_not_should_pass()
```
Comment (JeremySkinner) [inline src/FluentValidation.Tests/BooleanValidatorTests.cs]: Please can you re-word this test title? It doesn't make sense.

File: src/FluentValidation.Tests/BooleanValidatorTests.cs [hunk 12]
```diff
@@ -0,0 +1,108 @@
...
+		{
+			var validator = new TestValidator(v => v.RuleFor(x => x.NullableIsValid).True());
+			var result = validator.Validate(new Person { NullableIsValid = null });
+			result.IsValid.ShouldBeFalse();
+		}
+
+		[Fact]
+		public void When_the_false_boolean_nullable_value_validator_not_should_pass()
+		{
```
Comment (JeremySkinner) [inline src/FluentValidation.Tests/BooleanValidatorTests.cs]: Braces at end of line please.

File: src/FluentValidation.Tests/BooleanValidatorTests.cs [hunk 13]
```diff
@@ -0,0 +1,108 @@
...
+		[Fact]
+		public void When_the__true_boolean_nullable_value_validator_not_should_pass()
+		{
+			var validator = new TestValidator(v => v.RuleFor(x => x.NullableIsValid).True());
+			var result = validator.Validate(new Person { NullableIsValid = null });
+			result.IsValid.ShouldBeFalse();
+		}
+
+		[Fact]
+		public void When_the_false_boolean_nullable_value_validator_not_should_pass()
```
Comment (JeremySkinner) [inline src/FluentValidation.Tests/BooleanValidatorTests.cs]: Please reword this one too.

File: src/FluentValidation/DefaultValidatorExtensions.cs [hunk 14]
```diff
@@ -1057,5 +1057,58 @@ public static class DefaultValidatorExtensions {
 		public static IRuleBuilderInitial<T, TProperty> CustomAsync<T, TProperty>(this IRuleBuilder<T, TProperty> ruleBuilder, Func<TProperty, CustomContext, CancellationToken, Task> action) {
 			return (IRuleBuilderInitial<T, TProperty>) ruleBuilder.SetValidator(new CustomValidator<TProperty>(action));
 		}
+
+		/// <summary>
+		/// Defines a boolean validator on the current rule builder.
+		/// Validation will fail if the property is true.
```
Comment (JeremySkinner) [inline src/FluentValidation/DefaultValidatorExtensions.cs]: This is the wrong way around. For the `True` method, validation will fail if the property is `false`.

File: src/FluentValidation/DefaultValidatorExtensions.cs [hunk 15]
```diff
@@ -1057,5 +1057,58 @@ public static class DefaultValidatorExtensions {
...
+		/// <summary>
+		/// Defines a boolean validator on the current rule builder.
+		/// Validation will fail if the property is true.
+		/// </summary>
+		/// <typeparam name="T">Type of object being validated</typeparam>
+		/// <typeparam name="TProperty">Type of property being validated</typeparam>
+		/// <param name="ruleBuilder">The rule builder on which the validator should be defined</param>
+		/// <returns></returns>
+		public static IRuleBuilderOptions<T, bool> True<T>(this IRuleBuilder<T, bool> ruleBuilder)
+		{
```
Comment (JeremySkinner) [inline src/FluentValidation/DefaultValidatorExtensions.cs]: Braces at end of line please.

File: src/FluentValidation/DefaultValidatorExtensions.cs [hunk 16]
```diff
@@ -1057,5 +1057,58 @@ public static class DefaultValidatorExtensions {
...
+			return ruleBuilder.Equal(true);
+		}
+
+		/// <summary>
+		/// Defines a boolean validator on the current rule builder.
+		/// Validation will fail if the property is true.
```
Comment (JeremySkinner) [inline src/FluentValidation/DefaultValidatorExtensions.cs]: This is also the wrong way around. Should be "Validation will fail if the property is false or null."

File: src/FluentValidation/DefaultValidatorExtensions.cs [hunk 17]
```diff
@@ -1057,5 +1057,58 @@ public static class DefaultValidatorExtensions {
...
+		/// </summary>
+		/// <typeparam name="T">Type of object being validated</typeparam>
+		/// <typeparam name="TProperty">Type of property being validated</typeparam>
+		/// <param name="ruleBuilder">The rule builder on which the validator should be defined</param>
+		/// <returns></returns>
+		public static IRuleBuilderOptions<T, bool?> True<T>(this IRuleBuilder<T, bool?> ruleBuilder)
+		{
```
Comment (JeremySkinner) [inline src/FluentValidation/DefaultValidatorExtensions.cs]: Braces at end of line please.

File: src/FluentValidation/DefaultValidatorExtensions.cs [hunk 18]
```diff
@@ -1057,5 +1057,58 @@ public static class DefaultValidatorExtensions {
...
+			return ruleBuilder.Equal(true);
+		}
+
+		/// <summary>
+		/// Defines a boolean validator on the current rule builder.
+		/// Validation will fail if the property is false.
```
Comment (JeremySkinner) [inline src/FluentValidation/DefaultValidatorExtensions.cs]: This is the wrong way around. Should be "Validation will fail if the property is true."

File: src/FluentValidation/DefaultValidatorExtensions.cs [hunk 19]
```diff
@@ -1057,5 +1057,58 @@ public static class DefaultValidatorExtensions {
...
+		/// </summary>
+		/// <typeparam name="T">Type of object being validated</typeparam>
+		/// <typeparam name="TProperty">Type of property being validated</typeparam>
+		/// <param name="ruleBuilder">The rule builder on which the validator should be defined</param>
+		/// <returns></returns>
+		public static IRuleBuilderOptions<T, bool> False<T>(this IRuleBuilder<T, bool> ruleBuilder)
+		{
```
Comment (JeremySkinner) [inline src/FluentValidation/DefaultValidatorExtensions.cs]: Braces at end of line please.

File: src/FluentValidation/DefaultValidatorExtensions.cs [hunk 20]
```diff
@@ -1057,5 +1057,58 @@ public static class DefaultValidatorExtensions {
...
+			return ruleBuilder.Equal(false);
+		}
+
+		/// <summary>
+		/// Defines a boolean validator on the current rule builder.
+		/// Validation will fail if the property is false.
```
Comment (JeremySkinner) [inline src/FluentValidation/DefaultValidatorExtensions.cs]: This is the wroung way around. Should be "Validation will fail if the property is true or null".

File: src/FluentValidation/DefaultValidatorExtensions.cs [hunk 21]
```diff
@@ -1057,5 +1057,58 @@ public static class DefaultValidatorExtensions {
...
+		/// </summary>
+		/// <typeparam name="T">Type of object being validated</typeparam>
+		/// <typeparam name="TProperty">Type of property being validated</typeparam>
+		/// <param name="ruleBuilder">The rule builder on which the validator should be defined</param>
+		/// <returns></returns>
+		public static IRuleBuilderOptions<T, bool?> False<T>(this IRuleBuilder<T, bool?> ruleBuilder)
+		{
```
Comment (JeremySkinner) [inline src/FluentValidation/DefaultValidatorExtensions.cs]: Braces at end of line please.

File: src/FluentValidation/DefaultValidatorExtensions.cs [hunk 22]
```diff
@@ -1057,5 +1057,50 @@ public static class DefaultValidatorExtensions {
 		public static IRuleBuilderInitial<T, TProperty> CustomAsync<T, TProperty>(this IRuleBuilder<T, TProperty> ruleBuilder, Func<TProperty, CustomContext, CancellationToken, Task> action) {
 			return (IRuleBuilderInitial<T, TProperty>) ruleBuilder.SetValidator(new CustomValidator<TProperty>(action));
 		}
+
+		/// <summary>
+		/// Defines a boolean validator on the current rule builder.
+		/// Validation will fail if the property not is true.
```
Comment (JeremySkinner) [inline src/FluentValidation/DefaultValidatorExtensions.cs]: Should be "is not true" instead of "not is true"

File: src/FluentValidation/DefaultValidatorExtensions.cs [hunk 23]
```diff
@@ -1057,5 +1057,50 @@ public static class DefaultValidatorExtensions {
...
+		/// <typeparam name="T">Type of object being validated</typeparam>
+		/// <param name="ruleBuilder">The rule builder on which the validator should be defined</param>
+		/// <returns></returns>
+		public static IRuleBuilderOptions<T, bool> True<T>(this IRuleBuilder<T, bool> ruleBuilder) {
+			return ruleBuilder.Equal(true);
+		}
+
+		/// <summary>
+		/// Defines a boolean validator on the current rule builder.
+		/// Validation will fail if the property not is true.
```
Comment (JeremySkinner) [inline src/FluentValidation/DefaultValidatorExtensions.cs]: Should be "is not true" instead of "not is true"

File: src/FluentValidation/DefaultValidatorExtensions.cs [hunk 24]
```diff
@@ -1057,5 +1057,50 @@ public static class DefaultValidatorExtensions {
...
+		/// <typeparam name="T">Type of object being validated</typeparam>
+		/// <param name="ruleBuilder">The rule builder on which the validator should be defined</param>
+		/// <returns></returns>
+		public static IRuleBuilderOptions<T, bool?> True<T>(this IRuleBuilder<T, bool?> ruleBuilder) {
+			return ruleBuilder.Equal(true);
+		}
+
+		/// <summary>
+		/// Defines a boolean validator on the current rule builder.
+		/// Validation will fail if the property not is false.
```
Comment (JeremySkinner) [inline src/FluentValidation/DefaultValidatorExtensions.cs]: Should be "is not false" instead of "not is false"

File: src/FluentValidation/DefaultValidatorExtensions.cs [hunk 25]
```diff
@@ -1057,5 +1057,50 @@ public static class DefaultValidatorExtensions {
...
+		/// <typeparam name="T">Type of object being validated</typeparam>
+		/// <param name="ruleBuilder">The rule builder on which the validator should be defined</param>
+		/// <returns></returns>
+		public static IRuleBuilderOptions<T, bool> False<T>(this IRuleBuilder<T, bool> ruleBuilder)	{
+			return ruleBuilder.Equal(false);
+		}
+
+		/// <summary>
+		/// Defines a boolean validator on the current rule builder.
+		/// Validation will fail if the property not is false.
```
Comment (JeremySkinner) [inline src/FluentValidation/DefaultValidatorExtensions.cs]: Should be "is not false" instead of "not is false"

File: src/FluentValidation.Tests/BooleanValidatorTests.cs [hunk 26]
```diff
@@ -0,0 +1,97 @@
...
+
+		[Fact]
+		public void When_the_nullable_value_is_true_the_validator_not_should_pass()	{
+			var validator = new TestValidator(v => v.RuleFor(x => x.NullableIsValid).False());
+			var result = validator.Validate(new Person { NullableIsValid = true });
+			result.IsValid.ShouldBeFalse();
+		}
+
+		[Fact]
+		public void When_the__true_boolean_nullable_value_validator_not_should_pass() {
```
Comment (JeremySkinner) [inline src/FluentValidation.Tests/BooleanValidatorTests.cs]: Title of this test still doesn't make sense, please reword.

File: src/FluentValidation.Tests/BooleanValidatorTests.cs [hunk 27]
```diff
@@ -0,0 +1,97 @@
...
+			var validator = new TestValidator(v => v.RuleFor(x => x.NullableIsValid).True());
+			var result = validator.Validate(new Person { NullableIsValid = null });
+			result.IsValid.ShouldBeFalse();
+		}
+
+		[Fact]
+		public void When_the_false_boolean_nullable_value_validator_not_should_pass() {
```
Comment (JeremySkinner) [inline src/FluentValidation.Tests/BooleanValidatorTests.cs]: Title of this test still doesn't make sense, please reword.

**Comment by JeremySkinner** [comment]
I'm going to go ahead and close this as I feel the existing `Equal` overloads are sufficient.
